src/olympia/abuse/serializers.py

```diff
@@ -153,19 +154,30 @@ def to_internal_value(self, data):
             )
         self.validate_target(data, 'addon')
         view = self.context.get('view')
-        output = view.get_guid_and_addon()
-        # Pop 'addon' from data before passing that data to super(), we already
-        # have it in the output value.
+        output = {'guid': view.get_guid()}
+        # Pop 'addon' from data before passing that data to super(), we don't
+        # need it - we're only recording the guid and it's coming from the view
         data.pop('addon')
         output.update(super().to_internal_value(data))
         return output
 
     def get_addon(self, obj):
-        return {
+        output = {
             'guid': obj.guid,
-            'id': None,  # For backwards-compatibility.
-            'slug': None,  # For backwards-compatibility.
+            'id': None,
+            'slug': None,
         }
+        if view := self.context.get('view'):
```

Comment:
I'm not following how this is connected to the pr title - won't it always include the `pk` and `slug` if there's a view, irregardless of how the lookup was made?

Reply:
It would - if the add-on is public (so it shouldn't cause issues). I could have relied on the `view.addon_object` property we used for caching, but I thought it was a little too magical, and using `get_addon_object()` would be more consistent.

Reply:
so... this pr doesn't do what the issue/pr is titled (i.e. return the slug/id **if** it was used in the request) but just always returns the slug and id if the add-on is public?  And I guess it also wouldn't return the slug/id for a non-public add-on, even if the id/slug was used in the abuse report?

Should the issue be updated to reflect the change in functionality?  (And will this new operation still fix the issue on frontend?)

Reply:
Yes - essentially it reverts to the original behavior (except without storing the `addon` fk). I'll update the issue and PR title.